content/essentials/memos.mdx

```diff
@@ -0,0 +1,108 @@
+They, however, serve different purposes and each have their own unique behaviors.
+
+| | Memos  | Effects   |
+|-|--------|-----------|
+| Purpose | Derive and cache values based on reactive state.  | Handle side effects like DOM manipulations, network requests, or other external actions. |
+| Return value  | Yes - returns the result of the computation or derived state. | Does not return a value but executes a block of code in response to changes. |
+| Reactivity | Re-runs only when dependencies change. | Runs on every dependency change. |
+| Caches results | Yes | No |
+| Behavior | Designed to be pure functions without side effects. | Can cause side effects like UI updates or data fetches. |
+| Dependency tracking. | Automatically tracks all dependencies. | Automatically tracks all dependencies, running when relevant reactive values change. |
+| Example use cases | Transforming data structures, computing aggregated values, derived state, or other expensive computations. | UI updates, network requests, or external integrations. |
+
+## Best practices
+
+When working with memos, its recommended that you leave them "pure". 
+This means you should not modify other signals because it can lead to unexpected behaviors and make the code harder to maintain. 
+
+```jsx
+import { createSignal, createMemo } from 'solid-js';
+
+const [count, setCount] = createSignal(0);
+const isEven = createMemo(() => count() % 2 === 0); // example of a pure function
+```
+
+Since memos are expected to be pure functions, their outputs should solely depend on their inputs without side effects. 
+When you add a side effect, the reactivity chain is complicated which potentially introduces infinite update loops. 
+
+```jsx
+import { createSignal, createMemo } from 'solid-js';
+
+const [count, setCount] = createSignal(0);
+const [message, setMessage] = createSignal("");
+
+// impure function in createMemo
+const badMemo = createMemo(() => {
+    if (count() > 10) {
+        setMessage("Count is too high!"); // this can introduce a side effect
+    }
+    return count() % 2 === 0;
+});
+```
+
+In this example, if another part of the application reacts to the changes in the `message` signal, it can trigger further updates or side effects which complicate the flow of data and cause a possible infinite update loop. 
+
+If there's a need to perform a side effect based on a memo's value, it's better to use an effect:
+
+```jsx
+import { createSignal, createMemo, createEffect } from 'solid-js';
+
+const [count, setCount] = createSignal(0);
+const [message, setMessage] = createSignal("");
+
+const isEven = createMemo(() => count() % 2 === 0);
+
+createEffect(() => {
+    if (count() > 10) {
+        setMessage("Count is too high!");
+    }
+});
+```
+
+In the revised approach, the `isEven` memo will purely compute based on whether `count` is even *without* side effects. 
+In using the `createEffect` function, the side effects are appropriately handled. 
```

Comment:
Maybe worth saying something like: if the goal is to keep the message in sync with the count, to avoid multiple downstream updates, it's better still to write it as a memo:

```jsx
const message = createMemo(() => {
  if (count() > 10) {
    return "Count is too high!";
  } else {
    return "";
  }
})
```

I think showing this mentality of converting unnecessary effects (with side effects) into memos is a useful example. In general I think we want to say that memos are preferred when appropriate.

I wonder whether there should be a different, better example of side effects where effects are truly needed, like DOM updates or something.